### PR TITLE
[SD-1237] Display repository/current desk location in content views

### PR DESCRIPTION
**Pull request by akintolga**

### CODE REVIEW

File: app/scripts/superdesk-search/search.js [hunk 1]
```diff
@@ -546,6 +546,34 @@
             };
         }])
 
+        .directive('sdItemContainer', ['$location', '$filter', 'desks', 'api', function($location, $filter, desks, api) {
+            return {
+                scope: {
+                    item: '='
+                },
+                templateUrl: 'scripts/superdesk-search/views/item-container.html',
+                replace: true,
```
Comment (petrjasek) [inline app/scripts/superdesk-search/search.js]: replace is deprecated

File: app/scripts/superdesk-archive/views/media-box-grid.html [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
 <div class="media"  ng-show="item.type == 'text' || item.type == 'composite'">
     <small title="{{ item.headline }}">{{ item.headline }}</small>
+    <sd-item-container data-item="item" />
```
Comment (petrjasek) [inline app/scripts/superdesk-archive/views/media-box-grid.html]: better use `<div sd-item-container></div>` for browser compatibility and consistency

File: app/scripts/superdesk-search/search.js [hunk 3]
```diff
@@ -546,6 +546,34 @@
...
+                link: function(scope, elem) {
+
+                    if (!scope.item.task) {
+                        return;
+                    }
+
+                    if (scope.item.task.desk) {
+                        desks.initialize().then(function() {
+                            scope.item.container = 'on ' + desks.deskLookup[scope.item.task.desk].name ;
```
Comment (petrjasek) [inline app/scripts/superdesk-search/search.js]: `on ` can't be translated

File: app/scripts/superdesk-search/search.js [hunk 4]
```diff
@@ -546,6 +546,34 @@
...
+                        });
+                    } else if (scope.item.task.user) {
+                        api.find('users', scope.item.task.user).then(function(user) {
+                            scope.item.container = ' with ' + $filter('username')(user);
```
Comment (petrjasek) [inline app/scripts/superdesk-search/search.js]: `with` can't be translated

File: app/scripts/superdesk-search/search.js [hunk 5]
```diff
@@ -556,6 +556,31 @@
...
+                    if (!scope.item.task) {
+                        return;
+                    }
+
+                    if (scope.item.task.desk) {
+                        desks.initialize().then(function() {
+                            scope.item.container = 'desk:' + desks.deskLookup[scope.item.task.desk].name ;
+                        });
+                    } else if (scope.item.task.user) {
+                        scope.item.container = 'location:workspace';
```
Comment (akintolga) [inline app/scripts/superdesk-search/search.js]: @petrjasek Yesterday I was checking who the user is but that probably doesn't make sense. As everyone can see only their own private workspace, there's no point in checking the user. Plus it was a bit costly to do check for the results (esp. when 100 item is selected).

If this is needed in the future, I think we should add the username to the task schema as we probably don't expect the username for user_id would ever change

File: app/scripts/superdesk-search/search.js [hunk 6]
```diff
@@ -556,6 +556,31 @@
...
+                template: '{{item.container}}',
+                link: function(scope, elem) {
+
+                    if (!scope.item.task) {
+                        return;
+                    }
+
+                    if (scope.item.task.desk) {
+                        desks.initialize().then(function() {
+                            scope.item.container = 'desk:' + desks.deskLookup[scope.item.task.desk].name ;
```
Comment (akintolga) [inline app/scripts/superdesk-search/search.js]: I have changed to use desk: and location: for the time being as I think they would do the job (I hope this too would not create translation problem). I presume @nidzix would have better ideas how the location would be displayed as a later stage.

**Comment by petrjasek** [comment]
:+1:

**Comment by amagdas** [comment]
:+1:

**Pull request merged**